src/HtmlUtils.tsx

```diff
@@ -403,9 +404,11 @@ export function bodyToHtml(content: IContent, highlights: string[], opts: IOpts
     try {
         if (highlights && highlights.length > 0) {
             const highlighter = new HtmlHighlighter("mx_EventTile_searchHighlight", opts.highlightLink);
-            const safeHighlights = highlights.map(function(highlight) {
-                return sanitizeHtml(highlight, sanitizeParams);
-            });
+            const safeHighlights = highlights
+                // sanitizeHtml can hang if an unclosed HTML tag is thrown at it
+                // A search for `<foo` will make the browser crash
+                .filter((highlight: string): boolean => !highlight.includes("<"))
+                .map((highlight: string): string => sanitizeHtml(highlight, sanitizeParams));
```

Comment:
can we please not overload the imported name `highlight` - it confused me for longer than it should have

Reply:
hmm, can we not do any escaping here? simply not highlighting some things but highlighting others will feel broken I think

Reply:
This was my original thought, but with or without escaping the highlighting does not work for those special characters.

Also tested all the XML special characters `"`, `'`, `<`, `>`, `&` and they all work fine with `sanitizeHtml` except `<` that can make it explode in case of a malformed tag

That seemed like the most sensible approach as escaping will not improve the highlighting in any way

Reply:
then it at the very least needs an `XXX: ` comment to signify we know we are breaking things, but the alternative is worse